packages/salesforcedx-vscode-core/src/conflict/metadataCacheService.ts

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {
+  ComponentSet,
+  MetadataApiRetrieve,
+  RetrieveResult,
+  SourceComponent,
+  SourceRetrieveResult
+} from '@salesforce/source-deploy-retrieve';
+import { FileProperties } from '@salesforce/source-deploy-retrieve/lib/src/client/types';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as shell from 'shelljs';
+import * as vscode from 'vscode';
+import { RetrieveExecutor } from '../commands/baseDeployRetrieve';
+import { getRootWorkspacePath } from '../util';
+
+export interface MetadataContext {
+  baseDirectory: string;
+  commonRoot: string;
+  components: SourceComponent[];
+}
+
+export interface MetadataCacheResult {
+  selectedPath: string;
+  selectedIsDirectory: boolean;
+
+  cachePropPath?: string;
+  cache: MetadataContext;
+  project: MetadataContext;
+}
+
+export class MetadataCacheService {
+  private static CACHE_FOLDER = ['.sfdx', 'diff'];
+  private static PROPERTIES_FOLDER = ['prop'];
+  private static PROPERTIES_FILE = 'file-props.json';
+
+  private username: string;
+  private cachePath: string;
+  private componentPath?: string;
+  private projectPath?: string;
+  private sourceComponents: ComponentSet;
+
+  public constructor(username: string) {
+    this.username = username;
+    this.sourceComponents = new ComponentSet();
+    this.cachePath = this.makeCachePath(username);
+  }
+
+  public initialize(componentPath: string, projectPath: string): void {
+    this.componentPath = componentPath;
+    this.projectPath = projectPath;
+  }
+
+  public async loadCache(
+    componentPath: string,
+    projectPath: string
+  ): Promise<MetadataCacheResult | undefined> {
+    this.initialize(componentPath, projectPath);
+    const components = await this.getSourceComponents();
+    const operation = await this.createRetrieveOperation(components);
+    const results = await operation.start();
+    return this.processResults(results);
+  }
+
+  public async getSourceComponents(): Promise<ComponentSet> {
+    return this.componentPath
+      ? (this.sourceComponents = ComponentSet.fromSource(this.componentPath))
+      : new ComponentSet();
+  }
+
+  public async createRetrieveOperation(
+    comps?: ComponentSet
+  ): Promise<MetadataApiRetrieve> {
+    const components = comps || (await this.getSourceComponents());
+    this.clearDirectory(this.cachePath, true);
+
+    const operation = components.retrieve({
+      usernameOrConnection: this.username,
+      output: this.cachePath,
+      merge: false
+    });
+
+    return operation;
+  }
+
+  public async processResults(
+    result: RetrieveResult | SourceRetrieveResult | undefined
```

Comment:
I think we should be good to remove `SourceRetrieveResult` since the old tooling is not used anymore.